### PR TITLE
Validation ordering

### PR DESCRIPTION
**Pull request by jflo**
- corrects a number of validation errors by adjusting priority of evaluations
- covers and corrects a subtle issue where besu would propose bad blocks in the event of a re-org prior to a proposal which restored blob transactions to mempool.

### CODE REVIEW

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java [hunk 1]
```diff
@@ -469,6 +469,10 @@ && strictReplayProtectionShouldBeEnforcedLocally(chainHeadBlockHeader)
       return ValidationResultAndAccount.invalid(
           TransactionInvalidReason.INVALID_TRANSACTION_FORMAT,
           "EIP-1559 transaction are not allowed yet");
+    } else if (transaction.getType().equals(TransactionType.BLOB)
```
Comment (macfarla) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:472]: this is the change that fixed the invalid block proposals, right?

**Review by macfarla — APPROVED** [review]
looks pretty straight forward. couple of comments

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java (hunk 1 above)
Comment (jflo) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:472]: exactly! I considered updating a transaction validator to handle it, but decided against. it's not an invalid tx, it's just in the wrong context.

Comment (Gabriel-Trintinalia) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:472]: can we use transaction.getType().supportsBlob() here?

**Pull request merged**